panel/browser-panel.cpp

```diff
@@ -235,6 +235,70 @@ void QCefWidgetInternal::closeBrowser()
 	}
 }
 
+#ifdef __linux__
+static bool XWindowHasAtom(Display *display, Window w, Atom a)
+{
+	Atom type;
+	int format;
+	unsigned long nItems;
+	unsigned long bytesAfter;
+	unsigned char *data = NULL;
+
+	if (XGetWindowProperty(display, w, a, 0, LONG_MAX, False,
+			       AnyPropertyType, &type, &format, &nItems,
+			       &bytesAfter, &data) != Success)
+		return false;
+
+	if (data)
+		XFree(data);
+
+	return type != None;
+}
+
+void QCefWidgetInternal::unsetToplevelXdndProxy()
+{
+	if (!cefBrowser)
+		return;
+
+	CefWindowHandle browserHandle =
+		cefBrowser->GetHost()->GetWindowHandle();
+	Display *xDisplay = cef_get_xdisplay();
+	Window toplevel, root, parent, *children;
+	unsigned int nChildren;
+	bool found = false;
+
+	toplevel = browserHandle;
+
+	// Find the toplevel
+	Atom netWmPidAtom = XInternAtom(xDisplay, "_NET_WM_PID", False);
+	do {
+		if (XQueryTree(xDisplay, toplevel, &root, &parent, &children,
+			       &nChildren) == 0)
+			return;
```

Comment:
Am I misunderstanding or isnt this equivalent to `while(XQueryTree(xDisplay, toplevel, &root, &parent, &children, &nChildren) != 0) {}`

Reply:
The reason I was more strict with this check, and returned instead of just continuing or silently ignoring, is that when `XQueryTree()` returns zero, it usually means bad data was passed. In this case, we're literally feeding it its own data, so returning zero must be a socket error.

Also, all Xlib calls are blocking and potentially heavy, so I think it's better to stop calling them early.

Thanks to your comment, I noticed that `children` was being leaked in there! Updated with an appropriate XFree. Thanks!